Formula/git.rb

```diff
@@ -42,9 +44,13 @@ class Git < Formula
     sha256 "19e3cb0425c94d4ad82984f41522e77c8e35093e15a891f8e7195617201f6ac1"
   end
 
-  resource "Net::SMTP::SSL" do
-    url "https://cpan.metacpan.org/authors/id/R/RJ/RJBS/Net-SMTP-SSL-1.04.tar.gz"
-    sha256 "7b29c45add19d3d5084b751f7ba89a8e40479a446ce21cfd9cc741e558332a00"
+  use_ssl = Utils.safe_popen_read("perl", "-MNet::SMTP", "-e",
+                                  "print version->parse($Net::SMTP::VERSION) < version->parse('2.34')")
```

Comment:
```suggestion
  def use_ssl
    Utils.safe_popen_read("perl", "-MNet::SMTP", "-e",
                                  "print version->parse($Net::SMTP::VERSION) < version->parse('2.34')")
  end
```
Something like this should save having to define this twice. Please add a comment explaining what this is for.

Reply:
Errr, I tried below.

```ruby
  # git-send-email needs Net::SMTP::SSL or Net::SMTP >= 2.34
  def use_net_smtp_ssl?
    Utils.safe_popen_read("perl", "-MNet::SMTP", "-e",
                          "print version->parse($Net::SMTP::VERSION) >= version->parse('2.34')").empty?
  end

  if use_net_smtp_ssl?
    resource "Net::SMTP::SSL" do
      url "https://cpan.metacpan.org/authors/id/R/RJ/RJBS/Net-SMTP-SSL-1.04.tar.gz"
      sha256 "7b29c45add19d3d5084b751f7ba89a8e40479a446ce21cfd9cc741e558332a00"
    end
  end
```

But `brew install` failed:

```console
Error: git: undefined method `use_net_smtp_ssl?' for Formulary::FormulaNamespace53cdc2fd6a8c7040d167a58b9b57d045::Git:Class`
```

It seems that the defined method `use_net_smtp_ssl?` can not be used with the `resource` block. Am I mistaken somewhere? @carlocab

Reply:
OK, I define `use_net_smtp_ssl` as class methods.